plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastore.java

```diff
@@ -1076,6 +1103,70 @@ public void testDropTable()
         assertEquals(mockClient.getAccessCount(), 5);
     }
 
+    @Test
+    public void testAllDatabases()
+    {
+        assertThat(mockClient.getAccessCount()).isEqualTo(0);
+
+        assertThat(metastore.getAllDatabases()).containsExactly(TEST_DATABASE);
+        assertThat(mockClient.getAccessCount()).isEqualTo(1);
+        assertThat(metastore.getAllDatabases()).containsExactly(TEST_DATABASE);
+        assertThat(mockClient.getAccessCount()).isEqualTo(1); // should read it from cache
+
+        metastore.dropDatabase(TEST_DATABASE, false);
+
+        assertThat(metastore.getAllDatabases()).containsExactly(TEST_DATABASE);
+        assertThat(mockClient.getAccessCount()).isEqualTo(2);
+        assertThat(metastore.getAllDatabases()).containsExactly(TEST_DATABASE);
+        assertThat(mockClient.getAccessCount()).isEqualTo(2); // should read it from cache
+
+        metastore.createDatabase(
+                Database.builder()
+                        .setDatabaseName(TEST_DATABASE)
+                        .setOwnerName(Optional.empty())
+                        .setOwnerType(Optional.empty())
+                        .build());
+
+        assertThat(metastore.getAllDatabases()).containsExactly(TEST_DATABASE);
+        assertThat(mockClient.getAccessCount()).isEqualTo(3);
+        assertThat(metastore.getAllDatabases()).containsExactly(TEST_DATABASE);
+        assertThat(mockClient.getAccessCount()).isEqualTo(3); // should read it from cache
+    }
+
+    @Test
+    public void testAllTables()
+    {
+        assertThat(mockClient.getAccessCount()).isEqualTo(0);
+
+        assertThat(metastore.getAllTables()).contains(ImmutableList.of(new SchemaTableName(TEST_DATABASE, TEST_TABLE)));
```

Comment:
nit: extract variable for schema table name

Reply:
Extracted